gateway/modules/crud/mgo/read.go

```diff
@@ -69,8 +71,45 @@ func (m *Mongo) Read(ctx context.Context, col string, req *model.ReadRequest) (i
 			}
 		}
 
+		pipeline := make([]bson.M, 0)
+		if len(req.Aggregate) > 0 {
+			if len(req.Find) > 0 {
+				pipeline = append(pipeline, bson.M{"$match": req.Find})
+			}
+			for function, colArray := range req.Aggregate {
+				for _, column := range colArray {
+					asColumnName := generateAggregateAsColumnName(function, column)
+					switch function {
+					case "sum":
+						pipeline = generateQuery(pipeline, req, asColumnName, function, column)
+					case "min":
+						pipeline = generateQuery(pipeline, req, asColumnName, function, column)
+					case "max":
+						pipeline = generateQuery(pipeline, req, asColumnName, function, column)
+					case "avg":
+						pipeline = generateQuery(pipeline, req, asColumnName, function, column)
+					case "count":
+						pipeline = generateQuery(pipeline, req, asColumnName, function, "*")
+					default:
+						return 0, nil, utils.LogError(fmt.Sprintf(`Unknown aggregate funcion %s`, function), "mgo", "Read", nil)
+					}
+				}
+			}
```

Comment:
in this iteration only generation function map stage
after the outer for loop, generate the group by stage and add the function stage in group by stage